Dockerfile

```
@@ -1,21 +1,48 @@
+COPY proto ./proto
+COPY build.rs Cargo.toml Cargo.lock ./
+COPY src ./src
+
+RUN --mount=type=cache,id=cargo-registry,target=/usr/local/cargo/registry,sharing=locked \
+    --mount=type=cache,id=cargo-git,target=/usr/local/cargo/git,sharing=locked \
+    --mount=type=cache,id=cargo-target,target=/app/target,sharing=locked \
+    cargo build --release --locked && \
+    cp target/release/http2 /app/http2
+
+FROM debian:bookworm-slim AS runtime
+RUN apt-get update && \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)